src/commands/utils/errorable.ts

```diff
@@ -89,6 +89,16 @@ export function getErrorMessage(error: unknown) {
     return String(error);
 }
 
+export function getInnerErrorMessage(error: unknown) {
```

Comment:
I'm not sure we should be disabling the linting. Let me have a play around and try to make sense of this error structure. Unless it's documented somewhere?

Reply:
Okay, error types seem pretty poorly documented. The best I could find was [this](https://learn.microsoft.com/en-us/javascript/api/@azure/core-http/resterror?view=azure-node-latest#@azure-core-http-resterror-details), which simply lists the `details` property as being of type `unknown`.

Here is how I'd code it. It's a bit lengthy, but type-safe and descriptive:

```typescript
    } catch (ex) {
        const errorMessage = isInvalidTemplateDeploymentError(ex)
            ? getInvalidTemplateErrorMessage(ex)
            : getErrorMessage(ex);
        // ...
    }

function getInvalidTemplateErrorMessage(ex: InvalidTemplateDeploymentRestError): string {
    const innerDetails = ex.details.error?.details || [];
    if (innerDetails.length > 0) {
        const details = innerDetails.map((d) => `${d.code}: ${d.message}`).join("\n");
        return `Invalid template:\n${details}`;
    }

    const innerError = ex.details.error?.message || "";
    if (innerError) {
        return `Invalid template:\n${innerError}`;
    }

    return `Invalid template: ${getErrorMessage(ex)}`;
}

type InvalidTemplateDeploymentRestError = RestError & {
    details: {
        error?: {
            code: "InvalidTemplateDeployment";
            message?: string;
            details?: {
                code?: string;
                message?: string;
            }[];
        };
    };
};

function isInvalidTemplateDeploymentError(ex: unknown): ex is InvalidTemplateDeploymentRestError {
    return isRestError(ex) && ex.code === "InvalidTemplateDeployment";
}

function isRestError(ex: unknown): ex is RestError {
    return isObject(ex) && ex.constructor.name === "RestError";
}
```

Reply:
I'd put all that in `CreateClusterPanel.ts` and remove it from `errorable.ts`, because it's very specific to a particular function in one library, and is not generally applicable to other errors.

Reply:
@peterbom I expected this comments :),  I looked for the documentation for the types in the officials MS wiki's couldn't find one. I will look into your suggestion.

Reply:
@peterbom Thanks for suggestion, now the eslint warnings are done. I will test this and resolve this conversation.

Reply:
This was tested and now when the error happens, webview displays the correct inner exception details to the user. Thank you.